src/test/java/com/medallia/word2vec/Word2VecBinTest.java

```diff
@@ -0,0 +1,102 @@
+package com.medallia.word2vec;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
+import org.junit.Test;
+
+import com.medallia.word2vec.Searcher.UnknownWordException;
+import com.medallia.word2vec.util.Common;
+
+/**
+ * Tests converting the binary models into
+ * {@link com.medallia.word2vec.Word2VecModel}s.
+ * 
+ * @see com.medallia.word2vec.Word2VecModel#fromBinFile(File)
+ * @see com.medallia.word2vec.Word2VecModel#fromBinFile(File,
+ *      java.nio.ByteOrder)
+ */
+public class Word2VecBinTest {
+
+  /**
+   * Tests that the Word2VecModels created from a binary and text
+   * representations are equivalent
+   */
+  @Test
+  public void testRead()
+      throws IOException, UnknownWordException {
+    File binFile = Common.getResourceAsFile(
+            this.getClass(),
+            "/com/medallia/word2vec/tokensModel.bin");
+    Word2VecModel binModel = Word2VecModel.fromBinFile(binFile);
+
+    File txtFile = Common.getResourceAsFile(
+            this.getClass(),
+            "/com/medallia/word2vec/tokensModel.txt");
+    Word2VecModel txtModel = Word2VecModel.fromTextFile(txtFile);
+
+    assertModelsEqual(binModel, txtModel);
+  }
+
+  /**
+   * Tests that a Word2VecModel round-trips through the bin format without changes
+   */
+  @Test
+  public void testRoundTrip() throws IOException, UnknownWordException {
+    final String filename = "word2vec.c.output.model.txt";
+    final Word2VecModel model =
+        Word2VecModel.fromTextFile(filename, Common.readResource(Word2VecTest.class, filename));
+
+    final Path tempFile = Files.createTempFile(
+            String.format("%s-", Word2VecBinTest.class.getSimpleName()), ".bin");
+    try {
+      try (final OutputStream os = Files.newOutputStream(tempFile)) {
+        model.toBinFile(os);
+      }
+
+      final Word2VecModel modelCopy = Word2VecModel.fromBinFile(tempFile.toFile());
+      assertModelsEqual(model, modelCopy);
+    } finally {
+      Files.delete(tempFile);
```

Comment:
I don't like it because now the `tempFile` variable is declared far from where it's used, and there is state being carried from one function call to another function call, with a relationship that's not explicit in the code.

But it's your project, so I did it anyways :-)